builtin/credential/approle/validation.go

```diff
@@ -68,48 +68,34 @@ type secretIDAccessorStorageEntry struct {
 	SecretIDHMAC string `json:"secret_id_hmac" structs:"secret_id_hmac" mapstructure:"secret_id_hmac"`
 }
 
-// Checks if the Role represented by the RoleID still exists
-func (b *backend) validateRoleID(ctx context.Context, s logical.Storage, roleID string) (*roleStorageEntry, string, error) {
+// Validates the supplied RoleID and SecretID
+func (b *backend) validateCredentials(ctx context.Context, req *logical.Request, data *framework.FieldData) (*roleStorageEntry, string, map[string]string, string, error, error) {
```

Comment:
Consider naming the return parameters. The list is long, many are strings, and much of the usage is `nil, "",...` which makes it hard to infer what they even are.